src/main/java/org/folio/innreach/ccheduler/InitialContributionJobScheduler.java

```diff
@@ -0,0 +1,62 @@
+package org.folio.innreach.ccheduler;
+
+import com.google.common.cache.Cache;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.innreach.batch.contribution.service.InitialContributionEventProcessor;
+import org.folio.innreach.domain.entity.TenantInfo;
+import org.folio.innreach.domain.service.impl.TenantScopedExecutionService;
+import org.folio.innreach.repository.JobExecutionStatusRepository;
+import org.folio.innreach.repository.TenantInfoRepository;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+
+@Service
+@RequiredArgsConstructor
+@Log4j2
+public class InitialContributionJobScheduler {
+  private final TenantScopedExecutionService executionService;
+  private final JobExecutionStatusRepository jobExecutionStatusRepository;
+  private final InitialContributionEventProcessor eventProcessor;
+  private final TenantInfoRepository tenantRepository;
+  @Value(value = "${initial-contribution.fetch-limit}")
+  private int recordLimit;
+  @Value(value = "${initial-contribution.item-pause}")
+  private int itemPause;
+  private final Cache<String, List<String>> tenantDetailsCache;
+
+  @Scheduled(fixedDelayString = "${initial-contribution.scheduler.fixed-delay}",
+    initialDelayString = "${initial-contribution.scheduler.initial-delay}")
+  public void processInitialContributionEvents() {
+    List<String> tenants = loadTenants();
+    log.info("processInitialContributionEvents :: tenantsList {}", tenants);
+    tenants.forEach(tenant ->
+      executionService.runTenantScoped(tenant,
+        () -> {
+          try {
+            jobExecutionStatusRepository.updateAndFetchJobExecutionRecordsByStatus(recordLimit, itemPause)
```

Comment:
Just want to make sure I understand how concurrency is working for us here. Does a `recordLimit` of `2` mean we will have a concurrency level of 2, meaning two threads/cores will be working simultaneously. I believe this, paired with `@Async` on `processInitialContributionEvents` will have this effect but correct me if I'm wrong.

Reply:
Yes steve, you are correct. We will fetch `recordLimit ` number of records from the database and process every record asynchronously.